### PR TITLE
Add pyux test

### PR DESCRIPTION
**Pull request by farizrahman4u**
See #11992

### CODE REVIEW

File: tests/test_api.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import pytest
+import pyux
+import keras
+import json
+import os
+
+import keras.backend.tensorflow_backend
+import keras.backend.theano_backend
+import keras.backend.cntk_backend
+import keras.backend.numpy_backend
```
Comment (gabrieldemarmiesse) [inline tests/test_api.py]: Unused imports?

File: update_api.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import keras.backend.tensorflow_backend
+import keras.backend.theano_backend
+import keras.backend.cntk_backend
+import keras.backend.numpy_backend
+import keras.utils.test_utils
+
+sign = pyux.sign(keras)
+
+with open('api.json', 'w') as f:
+    json.dump(sign, f)
```
Comment (gabrieldemarmiesse) [inline update_api.py]: I would prefer the solution proposed by @fchollet . We can generate the file from master easily in travis by doing a git clone in another directory. In this way we never have to update the `api.json` file. Merging the branch will act as updating the `api.json` file so no manual step is required.

File: .travis.yml [hunk 3]
```diff
@@ -10,6 +10,8 @@ matrix:
           env: KERAS_BACKEND=tensorflow TEST_MODE=INTEGRATION_TESTS PIL=Pil
         - python: 3.6
           env: KERAS_BACKEND=tensorflow TEST_MODE=PEP8_DOC PIL=Pillow
+        - python: 3.6
+          env: KERAS_BACKEND=tensorflow TEST_MODE=API
```
Comment (gabrieldemarmiesse) [inline .travis.yml]: Is there a way to just add a label to the PR rather than just making the whole travis build fail?

File: update_api.py (hunk 2 above)
Comment (gabrieldemarmiesse) [reply update_api.py]: I believe we can do:
```shell
cd ../master_keras
git clone ...
cd keras
pip install .
generate_the_api_json_file
pip uninstall keras
cd ../../keras
```

This is easy to do with travis.ci but not for end users. I'm thinking about a way for end users to avoid changing their python installation.

Docker would maybe be a good fit or is it overkill?

Comment (gabrieldemarmiesse) [reply update_api.py]: @Dref360 I think your idea of having a github bot generating this file and put it somewhere on the web is a good idea.

Comment (farizrahman4u) [reply update_api.py]: > I believe we can do: `......`

This should work.

Comment (fchollet) [reply update_api.py]: The file `api.json` is very large (2.5M) so not a good fit for being checked in the git tree. The suggestion of @gabrieldemarmiesse where we add a script in Travis to do this work on startup sounds good to me.

Comment (Dref360) [reply update_api.py]: But then updating the file will be difficult and would require that you merge it. (Which I guess is something we want?)

The workflow would go like this:
1- User submits a PR with an API update
2- Travis fails because of this
3- fchollet has to force-merge the PR. 
4- The file is "updated" because it's computed on the most up-to-date master branch.

Comment (gabrieldemarmiesse) [reply update_api.py]: I looked into it and it seems quite complicated to do from travis. We should use a very descriptive job name and a very descriptive error to ensure that contributors don't assume that they have to change their code.
Something like TEST_MODE=API_DID_NOT_CHANGE . And with a descriptive error message from pyux and this behavior being documented in the `CONTRIBUTING.md`.

**Comment by farizrahman4u** [comment]
Closing PR as merging this would increase repo size by 3MB. Moved to  #12137.
